cmd/dns/cmd.go

```diff
@@ -49,9 +49,22 @@ func NewCmdValidateDns() *cobra.Command {
 			}
 
 			logger.Warn(ctx, "Using region: %s", config.region)
-			creds := credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
-			// The use of t3.micro here is arbitrary; we just need to provide any valid machine type
-			cli, err := cloudclient.NewClient(ctx, logger, creds, config.region, "t3.micro", nil)
+			var cli cloudclient.CloudClient
+			if config.awsProfile != "" || os.Getenv("AWS_ACCESS_KEY_ID") != "" {
+				// For AWS type
+				if config.awsProfile != "" {
+					logger.Info(ctx, "Using AWS profile: %s.", config.awsProfile)
+				} else {
+					logger.Info(ctx, "Using provided AWS credentials")
+				}
+				// The use of t3.micro here is arbitrary; we just need to provide any valid machine type
+				cli, err = cloudclient.NewClient(ctx, logger, config.region, "t3.micro", nil, "aws", config.awsProfile)
+
+			} else {
+				//	todo after GCP is implemented, check GCP type using creds
+				logger.Info(ctx, "GCP cloud credentials found.")
+				cli, err = cloudclient.NewClient(ctx, logger, config.region, "", nil, "gcp", config.awsProfile)
+			}
```

Comment:
This should be removed until we're ready to handle GCP credentials. Currently, this leads to the binary incorrectly sayings it using GCP credentials when its not, and passing in config.awsProfile, which is confusing.

Reply:
updated